shopfloor/actions/search.py

```diff
@@ -13,7 +13,9 @@ class SearchAction(Component):
     _usage = "search"
 
     def location_from_scan(self, barcode):
-        return self.env["stock.location"].search([("barcode", "=", barcode)])
+        return self.env["stock.location"].search(
+            [("barcode", "=", barcode), ("barcode", "!=", False)]
```

Comment:
wait, it means the argument `barcode` was empty, then we can avoid doing the SQL at all


```suggestion
        if not barcode:
            self.env["stock.location"].browse()
        return self.env["stock.location"].search([("barcode", "=", barcode)]
```

Reply:
fixed!